### PR TITLE
fix: eslint max depth warnings with passing E2E tests

### PR DESCRIPTION
**Pull request by samayer12**
This reverts commit 9dbf9ab9ab8b9781e623c12ef63834b22873b3aa.

## Description

#1249, but with attention paid to passing E2E tests that were missed in #1255. Two instances of `max-depth` violations remain in the codebase (`validate-processor.ts` & `mutate-processor.ts`), but that's a lot better than where things were.

End to End Test:  
(See [Pepr Excellent Examples](https://github.com/defenseunicorns/pepr-excellent-examples))

## Related Issue

Relates to #1249 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit, [Journey](https://github.com/defenseunicorns/pepr/tree/main/journey), [E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples), [docs](https://github.com/defenseunicorns/pepr/tree/main/docs), [adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or updated as needed
- [x] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

### CODE REVIEW

File: src/lib/logger.ts [hunk 1]
```diff
@@ -65,4 +65,16 @@ export function redactedPatch(patch: Record<string, Operation> = {}): Record<str
   return redactedCache;
 }
 
+export const logMutateErrorMessage = (e: Error): string => {
```
Comment (samayer12) [inline src/lib/logger.ts]: Internally, we've noted that logging needs a closer look (see #1360). So, to facilitate that process when we get there, I moved a log-related function to logger. Resolve this conversation if that's fine by the reviewer.

**Comment by samayer12** [comment]
> ## [Codecov](https://app.codecov.io/gh/defenseunicorns/pepr/pull/1366?
> ```diff
> @@            Coverage Diff             @@
> ##             main    #1366      +/-   ##
> ==========================================
> - Coverage   79.57%   79.53%   -0.04%     
> ==========================================
>   Files          39       39              
>   Lines        1816     1818       +2     
>   Branches      396      371      -25     
> ==========================================
> + Hits         1445     1446       +1     
> - Misses        369      370       +1     
>   Partials        2        2              
> ```

A -0.04% decrease in code-coverage while we refactor files that are large and hard to test in isolation at present isn't cause for concern that would prevent merging this PR.

File: src/lib/logger.ts (hunk 1 above)
Comment (samayer12) [reply src/lib/logger.ts]: We'll actually keep this in `mutate-processor.ts` until we address #1360

**Review by btlghrants — APPROVED** [review]
Seems to do the job to me.  The codecov drop isn't really an issue -- there's going to be some churn on this in the next few stories too, likely.

**Pull request merged**